formats/filtered/src/Filtered.php

```diff
@@ -164,6 +165,7 @@ protected function getResultText( SMWQueryResult $res, $outputmode ) {
 		$result = [];
 		while ( $row = $res->getNext() ) {
 			$result[ $this->uniqid() ] = new ResultItem( $row, $this );
+			usleep( 1 );
```

Comment:
> On a result array of 50 results I get sometimes 8, sometimes 20-ish or 30-ish items, depending on the loop design.

You mean in `SMWQueryResult` the `next` is returning different results for the same query? Have you tried to verify that `mResults` contains the same result before `next` is applied?

```
	public function getNext() {
		$page = current( $this->mResults );
		next( $this->mResults );
		...
```

`mResults` itself is a simple array (not an iterator nor a generator) so that should not cause any pointer issues.